### DISCUSSION TITLE
Accessing "unknown keys" and their values from `refine()` without `passthrough()`

### DISCUSSION THREAD

**Discussion by Aaronius** [colinhacks/zod#1029] [Q&A] [answered]
I have an object like so:

```
{
  a: "foo",
  b: "foo"
}
```

which I'm validating using a schema like this:

```
z.object({
  a: z.string()
}).refine((input) => {
  // I need to reference b here
});
```

I can access `b` if I call `passthrough()` before `refine()`, like this:

```
z.object({
  a: z.string()
})
.passthrough()
.refine((input) => {
  // I need to reference b here
});
```

I don't want `b` to actually be passed through though. Is there a way I can access `b` in `refine` without having to call `passthrough`?

**Comment by scotttrinh** [comment, marked as answer]
I would suggest adding the `b` schema and using `refine` to do your check and then `transform` to only return `a`.

**Reply by Aaronius** [reply]
That's workable. Thanks. It seems like a better solution would be to allow this to be configured through `RefineParams`. I think I'll open an issue for that.
